File: app/src/main/java/amrabed/android/release/evaluation/data/models/DayViewModel.java
package amrabed.android.release.evaluation.data.models;

import android.app.Application;

import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;

import java.util.List;

import amrabed.android.release.evaluation.data.Repository;
import amrabed.android.release.evaluation.data.entities.Day;

public class DayViewModel extends AndroidViewModel {
    private final Repository repository;
    private final LiveData<List<Day>> dayList;
    private MutableLiveData<Day> selectedDay = new MutableLiveData<>();

    public DayViewModel(Application application) {
        super(application);
        repository = new Repository(application);
        dayList = repository.loadAllDays();
    }

    public LiveData<List<Day>> getDayList() {
        return dayList;
    }

    public void select(Day day) {
        selectedDay.setValue(day);
    }

    public LiveData<Day> getSelected() {
        return selectedDay;
    }

    public void updateDay(Day day) {
        repository.updateDay(day);
    }
}

File: app/src/main/java/amrabed/android/release/evaluation/data/models/DayViewModel.kt
package amrabed.android.release.evaluation.data.models

import amrabed.android.release.evaluation.data.entities.Day
import amrabed.android.release.evaluation.data.repositories.DayRepository
import android.app.Application
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData

class DayViewModel(application: Application) : AndroidViewModel(application) {
    private val repository: DayRepository = DayRepository(application)
    val dayList: LiveData<List<Day?>?>? by lazy {
        repository.loadAllDays()
    }
    private val selectedDay = MutableLiveData<Day?>()

    fun select(day: Day?) {
        selectedDay.value = day
    }

    val selected: LiveData<Day?>
        get() = selectedDay

    fun updateDay(day: Day?) {
        repository.updateDay(day)
    }
}